### DISCUSSION TITLE
Accessing a variable contained in the state

### DISCUSSION THREAD

**Discussion by paul-stork** [reflex-dev/reflex#3065] [Q&A] [answered]
Good day!

I have a question for those who are much smarter than I...

I'm currently working on a proof of concept app that accesses 3 API's that all use the same set of credentials.

I currently get the credentials from a form and then pass them to:
```
class UserInfo(rx.Base):
    current_user: str
    user_pass: str

class FormState(rx.State):

    current_user_info: UserInfo = UserInfo(
        current_user='',
        user_pass = ''
    )
```

What I'm stuck on is figuring out how to use the credentials stored in the UserInfo base to authenticate the other two APIs on other pages without asking for their credentials over and over. 

While I would prefer to save a JWT, and use that to authenticate: I can't right now.

Long story short: I need to access something stored in Base or State and be able to use it in my Python code to authenticate an API call and I'm lost.

**Comment by picklelo** [comment, marked as answer]
When do you want to validate them, when they submit the form? I recommend going through the [local auth app](https://github.com/reflex-dev/reflex-examples/tree/8ed4a74023e71361596c2c1fad24e151f1c4c92f/local_auth). You should be able to authenticate in an event handler.

```python
class FormState(rx.State):

    current_user_info: UserInfo = UserInfo(
        current_user='',
        user_pass = ''
    )
    
    def authenticate(self):
        # Call your api here
        is_valid = my_api(self.current_user, self.user_pass)
```

You can then call the `authenticate` event handler whenever you want to run your API call.

**Reply by paul-stork** [reply]
I'll have to give this a try. Thanks!

I'm still trying to wrap my brain around all of this and I'm very excited about this project.
